File: CODIGO7/exercise.dart
import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync());
  int b = int.parse(stdin.readLineSync());

  int luckyTickets = 0;
  for (int i = 0; i < (1 << n) - 1; i++) {
    int firstHalf = 0;
    int secondHalf = 0;
    int j = 0;
    while (i > 0) {
      int digit = i % b;
      firstHalf += digit if j < n / 2 else secondHalf += digit;
      i >>= 1;
      j += 1;
    }
    if (firstHalf == secondHalf) {
      luckyTickets++;
    }
  }

  print(luckyTickets);
}

File: CODIGO7/exercise.rs
fn lucky_tickets(n: usize, b: usize) -> usize {
    let mut lucky_tickets = 0;
    let max_value = (1 << n) - 1;
    for i in 0..max_value {
      let mut first_half = 0;
      let mut second_half = 0;
      let mut j = 0;
      while i > 0 {
        let digit = i % b;
        first_half += digit if j < n / 2 else second_half += digit;
        i >>= 1;
        j += 1;
      }
      if first_half == second_half {
        lucky_tickets += 1;
      }
    }
    return lucky_tickets;
  }
  
  fn main() {
    let n = read_line().unwrap().parse().unwrap();
    let b = read_line().unwrap().parse().unwrap();
    println!("{}", lucky_tickets(n, b));
  }